Cassandra 4.0 (#1078)
mulesoft/docs-connectors: cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc

= Apache Cassandra Connector 4.0 Reference - Mule 4
:page-aliases: connectors::cassandra/cassandra-connector-reference.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

Anypoint Connector for Cassandra (Cassandra Connector) implements the DataStax Java Driver for Apache Cassandra 3.1.

> Release Notes: xref:release-notes::connector/cassandra-connector-release-notes-mule-4.adoc[Cassandra Connector Release Notes]

Reviewer: ```suggestion

```

Reviewer: This is in the box of the Connectors Landing page.

* CASSANDRA-DB:INVALID_TLS_CONFIGURATION
* CASSANDRA-DB:INVALID_TYPE
* CASSANDRA-DB:NO_HOST_AVAILABLE
* CASSANDRA-DB:OPERATION_TIMED_OUT
* CASSANDRA-DB:OVERLOADED
* CASSANDRA-DB:PAGING_STATE
* CASSANDRA-DB:PROTOCOL_ERROR
* CASSANDRA-DB:QUERY_CONSISTENCY
* CASSANDRA-DB:QUERY_ERROR
* CASSANDRA-DB:READ_FAILURE
* CASSANDRA-DB:READ_TIMEOUT
* CASSANDRA-DB:RETRY_EXHAUSTED
* CASSANDRA-DB:SERVER_ERROR
* CASSANDRA-DB:SYNTAX_ERROR
* CASSANDRA-DB:TRACE_RETRIEVAL
* CASSANDRA-DB:TRANSPORT
* CASSANDRA-DB:TRUNCATE
* CASSANDRA-DB:UNAUTHORIZED
* CASSANDRA-DB:UNAVAILABLE
* CASSANDRA-DB:UNPREPARED
* CASSANDRA-DB:UNRESOLVED_USER_TYPE
* CASSANDRA-DB:UNSUPPORTED_FEATURE
* CASSANDRA-DB:UNSUPPORTED_PROTOCOL_VERSION
* CASSANDRA-DB:WRITE_FAILURE
* CASSANDRA-DB:WRITE_TIMEOUT

[[dropColumn]]
=== Drop Column
`<cassandra-db:drop-column>`

Removes a column

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Table a| String |  The name of the table to use for the operation |  | x
| Keyspace Name a| String |  The keyspace that contains the table to use |  |

> | Column Name a| String |  The name of the column to removed |  #[payload] |

Reviewer: ```suggestion
| Column Name a| String |  The name of the column to remove |  #[payload] |
```

* CASSANDRA-DB:FRAME_TOO_LONG
* CASSANDRA-DB:FUNCTION_EXECUTION
* CASSANDRA-DB:INVALID_CONFIGURATION_IN_QUERY
* CASSANDRA-DB:INVALID_QUERY
* CASSANDRA-DB:INVALID_TLS_CONFIGURATION
* CASSANDRA-DB:INVALID_TYPE
* CASSANDRA-DB:NO_HOST_AVAILABLE
* CASSANDRA-DB:OPERATION_TIMED_OUT
* CASSANDRA-DB:OVERLOADED
* CASSANDRA-DB:PAGING_STATE
* CASSANDRA-DB:PROTOCOL_ERROR
* CASSANDRA-DB:QUERY_CONSISTENCY
* CASSANDRA-DB:QUERY_ERROR
* CASSANDRA-DB:READ_FAILURE
* CASSANDRA-DB:READ_TIMEOUT
* CASSANDRA-DB:RETRY_EXHAUSTED
* CASSANDRA-DB:SERVER_ERROR
* CASSANDRA-DB:SYNTAX_ERROR
* CASSANDRA-DB:TRACE_RETRIEVAL
* CASSANDRA-DB:TRANSPORT
* CASSANDRA-DB:TRUNCATE
* CASSANDRA-DB:UNAUTHORIZED
* CASSANDRA-DB:UNAVAILABLE
* CASSANDRA-DB:UNPREPARED
* CASSANDRA-DB:UNRESOLVED_USER_TYPE
* CASSANDRA-DB:UNSUPPORTED_FEATURE
* CASSANDRA-DB:UNSUPPORTED_PROTOCOL_VERSION
* CASSANDRA-DB:WRITE_FAILURE
* CASSANDRA-DB:WRITE_TIMEOUT

[[update]]
=== Update  Entity
`<cassandra-db:update>`

Executes the update entity operation

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x

> | Table a| String |  The name of the table in which the entity is updated |  | x

Reviewer: ```suggestion
| Table a| String |  The name of the table in which to update the entity |  | x
```
